### PR TITLE
[DYN-2772] Disable IronPython Alerts by Selection

### PR DESCRIPTION
**Pull request by zeusongit**
### Purpose

To enable users to hide IronPython alerts.

- Added 'Do not show again' checkbox to IronPython alerts.
- Added global switch option to turn off/on IronPython alerts, this setting is preserved for the user.
- Added tests

**Note:** You will probably need to enable `Python2ObsoleteMode` debug mode **before** starting the Dynamo Sandbox by editing the `debug.config` file.

SS:
Screen Shot 2020-07-17 at 12 01 41 PM

### Declarations

Check these if you believe they are true

- [x] The codebase is in a better state after this PR
- [x] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [x] The level of testing this PR includes is appropriate
- [x] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [x] Snapshot of UI changes, if any.

### Reviewers

@DynamoDS/dynamo

### CODE REVIEW

File: src/PythonMigrationViewExtension/IronPythonInfoDialog.xaml.cs [hunk 1]
```diff
@@ -68,6 +68,14 @@ private void OnMoreInformationButtonClicked(object sender, RoutedEventArgs e)
             this.Close();
         }
 
+        private void ToggleIronPythonDialog(object sender, RoutedEventArgs e)
+        {
+            if(DisableIronPythonDialogCheck.IsChecked.HasValue && DisableIronPythonDialogCheck.IsChecked.Value)
+                ViewModel.DynamoViewModel.IsIronPythonDialogDisabled = true;
+            else
+                ViewModel.DynamoViewModel.IsIronPythonDialogDisabled = false;
```
Comment (QilongTang) [inline src/PythonMigrationViewExtension/IronPythonInfoDialog.xaml.cs:76]: Could it be null? If not, how about
```
ViewModel.DynamoViewModel.IsIronPythonDialogDisabled = DisableIronPythonDialogCheck.IsChecked;
```

File: src/PythonMigrationViewExtension/PythonMigrationViewExtension.cs [hunk 2]
```diff
@@ -71,8 +71,8 @@ public void Loaded(ViewLoadedParams p)
 
         private void DisplayIronPythonDialog()
         {
-            // we only want to create the dialog ones for each graph per Dynamo session
-            if (DialogTracker.ContainsKey(CurrentWorkspace.Guid))
+            // we only want to create the dialog ones for each graph per Dynamo session, if the global setting is not disabled
+            if (DialogTracker.ContainsKey(CurrentWorkspace.Guid) || DynamoViewModel.IsIronPythonDialogDisabled)
```
Comment (QilongTang) [inline src/PythonMigrationViewExtension/PythonMigrationViewExtension.cs:75]: ah.. Not really related to this task but this line might be the reason why we have bug for dyf everytime we open..

Comment (mjkkirschner) [reply src/PythonMigrationViewExtension/PythonMigrationViewExtension.cs:75]: yes, tracking guid and not function id?

Comment (QilongTang) [reply src/PythonMigrationViewExtension/PythonMigrationViewExtension.cs:75]: @mjkkirschner yup

Comment (zeusongit) [reply src/PythonMigrationViewExtension/PythonMigrationViewExtension.cs:75]: Do you mean this bug?
https://jira.autodesk.com/browse/DYN-2936

File: src/PythonMigrationViewExtension/IronPythonInfoDialog.xaml.cs (hunk 1 above)
Comment (zeusongit) [reply src/PythonMigrationViewExtension/IronPythonInfoDialog.xaml.cs]: keeping the if in place, just in case

Comment (QilongTang) [reply src/PythonMigrationViewExtension/IronPythonInfoDialog.xaml.cs]: With the new code, it seems we can never update the value when `DisableIronPythonDialogCheck.IsChecked.Value` is `false`? Can you verify?

Comment (zeusongit) [reply src/PythonMigrationViewExtension/IronPythonInfoDialog.xaml.cs]: Ah.. sorry, fixed

File: src/PythonMigrationViewExtension/IronPythonInfoDialog.xaml.cs [hunk 3]
```diff
@@ -68,6 +68,11 @@ private void OnMoreInformationButtonClicked(object sender, RoutedEventArgs e)
             this.Close();
         }
 
+        private void ToggleIronPythonDialog(object sender, RoutedEventArgs e)
+        {
+            ViewModel.DynamoViewModel.IsIronPythonDialogDisabled = DisableIronPythonDialogCheck.IsChecked.Value;
```
Comment (QilongTang) [inline src/PythonMigrationViewExtension/IronPythonInfoDialog.xaml.cs:73]: 👍

File: src/PythonMigrationViewExtension/PythonMigrationViewExtension.cs (hunk 2 above)
Comment (QilongTang) [reply src/PythonMigrationViewExtension/PythonMigrationViewExtension.cs:75]: Nope this one: https://jira.autodesk.com/browse/DYN-2937. Anyway, I think out of scope..

**Pull request merged**